pkg/system/autoscaler.go

```diff
@@ -493,22 +495,23 @@ func (r *Reconciler) reconcilePrometheusAdapterResources(prometheus *monitoringv
 	if !util.KubeCreateSkipExisting(adapterAPIService) {
 		return fmt.Errorf("Error while crateiing APIService prometheus-adapter")
 	}
-	adapterHPA := util.KubeObject(bundle.File_deploy_internal_hpav2_autoscaling_yaml).(*autoscalingv2.HorizontalPodAutoscaler)
+	if err := r.ReconcileObject(r.AdapterHPA, r.reconcileAdapterHPA); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *Reconciler) reconcileAdapterHPA() error {
 	endpointsSpec := r.NooBaa.Spec.Endpoints
 	var minReplicas int32 = 1
 	var maxReplicas int32 = 2
 	if endpointsSpec != nil {
-		adapterHPA.Spec.MinReplicas = &endpointsSpec.MinCount
-		adapterHPA.Spec.MaxReplicas = endpointsSpec.MaxCount
-	} else {
-		adapterHPA.Spec.MinReplicas = &minReplicas
-		adapterHPA.Spec.MaxReplicas = maxReplicas
-	}
-	adapterHPA.Namespace = r.Request.Namespace
-	if err := r.ReconcileObject(adapterHPA, nil); err != nil {
-		return err
+		minReplicas = endpointsSpec.MinCount
+		maxReplicas = endpointsSpec.MaxCount
 	}
-
+	r.AdapterHPA.Spec.MinReplicas = &minReplicas
+	r.AdapterHPA.Spec.MaxReplicas = maxReplicas
+	r.AdapterHPA.Spec.Metrics[0].Object.Target.Value = nil
```

Comment:
@shirady without this change operator will simply create the HPA only once and do not update
`target.value` should set to nil otherwise HPA reconciler (in Kubernetes code) will try to validate the code

Reply:
comment added

Reply:
Thanks